src/Symfony/Component/Lock/Lock.php

```diff
@@ -124,6 +125,8 @@ public function refresh($ttl = null)
             }
 
             $this->logger->info('Expiration defined for "{resource}" lock for "{ttl}" seconds.', array('resource' => $this->key, 'ttl' => $ttl));
+        } catch (NotExpirableStoreException $e) {
+            $this->logger->notice('The store does not support expiration of locks.', array('store' => $this->store));
```

Comment:
The method `refresh` is called internally when calling `acquire` with a non-null TTL (which is the default case when using the Factory with default arguments)

When using a NotExpirableStore (like flock) the logger will be full of notice.
Same comment when combining a non-expirable store and expirable one (flock + redis for instance)

IMHO You should either silent this exception when calling `acquire` or update the factory to set a null TTL when the store does not support expiration.

Would be great from a user point of view to known whether or not the store supports expiration.
```
interface StoreInterface {
  public function save(Key $key);
  public function waitAndSave(Key $key);
  public function delete(Key $key);
  public function exists(Key $key);
}

interface ExpirableStoreInterface extends StoreInterface {
  public function supportsExpiration(): bool
  public function putOffExpiration(Key $key, $ttl);
}
```